### PR TITLE
[MM-28656] Automate release process

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR implements the required steps to (mostly) automate the release process.
It adds a `release` target to the `Makefile` that will update the version strings in the code, commit, add a new git tag and finally call [goreleaser](https://goreleaser.com/) to handle build, packaging and publishing to Github.

All this process should now be automated by a single command, e.g.:
```NEXT_VER=v1.1.1 make release```.

#### Ticket

https://mattermost.atlassian.net/browse/MM-28656

### CODE REVIEW

**Comment by streamer45** [comment]
Next step would be to have all this run in the CI but I think this is already an improvement.

File: Makefile [hunk 1]
```diff
@@ -95,6 +95,28 @@ golangci-lint:
...
+		if [[ "${TAG_EXISTS}" -eq 0 ]]; then \
+		  echo "Error: tag ${NEXT_VER} already exists"; \
+			exit -1; \
+		else \
+			for file in $(shell grep -rPl --include="*.go" --include="*.json" $(MATCH)); do \
+			sed -r -i 's/$(MATCH)/$(REPLACE)/g' $$file; \
+			done; \
+			git commit -a -m 'Releasing $(NEXT_VER)'; \
+			git tag $(NEXT_VER); \
+			goreleaser --rm-dist; \
```
Comment (agnivade) [inline Makefile:116]: Please also add a check to confirm that goreleaser is installed, with a link to the install page if the command is not found.

**Review by agnivade — COMMENTED** [review]
Makefile supports args. I guess it should also run with "make release NEXT_VER=<>"? And let's also document this step in the README.

**Comment by streamer45** [comment]
> Makefile supports args. I guess it should also run with "make release NEXT_VER=<>"? 

Yes that would work too.

> And let's also document this step in the README.

I think it makes sense to document this but maybe a dedicated documentation file would be better since this is an information needed only by core developers.

File: Makefile (hunk 1 above)
Comment (streamer45) [reply Makefile:116]: Sounds good :+1:

**Comment by agnivade** [comment]
> I think it makes sense to document this but maybe a dedicated documentation file would be better since this is an information needed only by core developers.

Makes sense. :+1:

File: Makefile [hunk 2]
```diff
@@ -95,6 +95,28 @@ golangci-lint:
...
+
+release:
+	@if [[ -z "${NEXT_VER}" ]]; then \
+		echo "Error: NEXT_VER must be defined"; \
+		exit -1; \
+	else \
+		if [[ "${TAG_EXISTS}" -eq 0 ]]; then \
+		  echo "Error: tag ${NEXT_VER} already exists"; \
+			exit -1; \
+		else \
```
Comment (isacikgoz) [inline Makefile:110]:
```suggestion
		else \
			if ! [ -x "$(command -v goreleaser)" ]; then \
			echo "goreleaser is not installed, do you want to download it? [y/N] " && read ans && [ $${ans:-N} = y ]; \
				if [ $$ans = y ] || [ $$ans = Y ]  ; then \
					curl -sfL https://install.goreleaser.com/github.com/goreleaser/goreleaser.sh | sh; \
				else \
					echo "aborting make release."; \
					exit -1; \
				fi; \
			fi; \
```

File: Makefile (hunk 1 above)
Comment (isacikgoz) [reply Makefile:116]: 😎

**Review by isacikgoz — APPROVED** [review]
Looks good to me 👍

File: Makefile (hunk 2 above)
Comment (streamer45) [reply Makefile:110]: That was easy :stuck_out_tongue:

**Review by agnivade — APPROVED** [review]

**Pull request merged**